File: flutter_application_2/backend/app/models/lesson_plan.py
from pydantic import BaseModel, Field
from typing import List, Optional

class LessonSection(BaseModel):
    """
    教学环节模型
    """
    title: str = Field(..., description="环节标题")
    content: str = Field(..., description="环节内容")
    duration_minutes: int = Field(..., description="时长（分钟）")
    activity_type: Optional[str] = Field(None, description="活动类型")
    resources: Optional[List[str]] = Field(default_factory=list, description="相关资源ID列表")

class LessonPlan(BaseModel):
    """
    教案模型
    """
    id: str = Field(..., description="教案唯一ID")
    title: str = Field(..., description="教案标题")
    subject: str = Field(..., description="学科")
    grade: str = Field(..., description="年级")
    duration_minutes: int = Field(..., description="总时长")
    sections: List[LessonSection] = Field(..., description="教学环节列表")
    objectives: Optional[str] = Field(None, description="教学目标")
    materials: Optional[str] = Field(None, description="教学材料")
    assessment: Optional[str] = Field(None, description="评估方式")
    notes: Optional[str] = Field(None, description="教学笔记")
    is_favorite: bool = Field(default=False, description="是否收藏")
    author_id: Optional[str] = Field(None, description="创建者ID")
    created_at: Optional[str] = Field(None, description="创建时间")
    updated_at: Optional[str] = Field(None, description="更新时间")

File: flutter_application_2/lib/models/lesson_plan.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:uuid/uuid.dart';

part 'lesson_plan.freezed.dart';
part 'lesson_plan.g.dart';

/// 教案模型类，用于表示AI生成的教案内容
@freezed
class LessonPlan with _$LessonPlan {
  const factory LessonPlan({
    required String id,
    required String title,
    required String subject,
    required String grade,
    required int durationMinutes,
    required List<LessonSection> sections,
    String? objectives,
    String? materials,
    String? assessment,
    String? notes,
    @Default(false) bool isFavorite,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) = _LessonPlan;

  /// 创建新的教案实例
  factory LessonPlan.create({
    required String title,
    required String subject,
    required String grade,
    required int durationMinutes,
    required List<LessonSection> sections,
    String? objectives,
    String? materials,
    String? assessment,
    String? notes,
  }) {
    return LessonPlan(
      id: const Uuid().v4(),
      title: title,
      subject: subject,
      grade: grade,
      durationMinutes: durationMinutes,
      sections: sections,
      objectives: objectives,
      materials: materials,
      assessment: assessment,
      notes: notes,
      createdAt: DateTime.now(),
      updatedAt: DateTime.now(),
    );
  }

  factory LessonPlan.fromJson(Map<String, dynamic> json) =>
      _$LessonPlanFromJson(json);
}

/// 教案章节模型，表示教案中的一个教学环节
@freezed
class LessonSection with _$LessonSection {
  const factory LessonSection({
    required String title,
    required String content,
    required int durationMinutes,
    String? activityType,
    List<String>? resources,
  }) = _LessonSection;

  factory LessonSection.fromJson(Map<String, dynamic> json) =>
      _$LessonSectionFromJson(json);
} 